realn.c

```diff
@@ -106,10 +106,29 @@ static int realn_check_tag(const uint8_t *tg, enum htsLogLevel severity,
 int sam_prob_realn(bam1_t *b, const char *ref, hts_pos_t ref_len, int flag)
 {
     int k, bw, y, yb, ye, xb, xe, apply_baq = flag&1, extend_baq = flag>>1&1, redo_baq = flag&4, fix_bq = 0;
+    enum htsRealnPlatform system = (flag>>3)<<3;
```

Comment:
I think this might be better as `system = flag & (0x1f << 3)` to select the exact bits mentioned in the header.

Reply:
I thought about that.  My comment hypothecated it being bits 3-7, but we don't use anything higher so it makes zero difference so I just use whatever is remaining to provide freedom for changing things.

I question whether this amendment of enum values really matters anyway.  We're trading having to do shifting in the caller (which admittedly doesn't even exist at the moment), vs having to do shifting in the htslib side.  Given we already have a bit field where the user is expected to shift-or, shift-or, shift-or, it may even be counterintuitive to require one and only one element to not be shifted.